config/config.go

```diff
@@ -117,6 +118,18 @@ func ObjectIDForKey(key roachpb.RKey) (uint32, bool) {
 	return uint32(id64), err == nil
 }
 
+// Hash returns a SHA1 hash of the SystemConfig contents.
+func (s SystemConfig) Hash() []byte {
+	sha := sha1.New()
+	for _, kv := range s.Values {
+		_, _ = sha.Write(kv.Key)
+		// There are all kinds of different types here, so we can't use the typed
+		// getters.
+		_, _ = sha.Write(kv.Value.RawBytes)
+	}
+	return sha.Sum(nil)
+}
```

Comment:
This is movement of code from `Replica.loadSystemDBSpan` so that I could generate the `SystemConfig` hash in other places while debugging #3697.